### PR TITLE
[Keymap] add crkbd/keymaps/markstos

### PR DESCRIPTION
**Pull request by markstos**
## Description

Adds "markstos" layout for crkbd

## Types of Changes

- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/markstos/config.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+// will perform its tap action. If the HOLD_ON_OTHER_KEY_PRESS option is
+// enabled, the dual-role key will perform its hold action instead.
+#define HOLD_ON_OTHER_KEY_PRESS
+
+// markstos: prevent keydown and keyup from firing on different layers
+#define PREVENT_STUCK_MODIFERS
+
+#ifdef RGBLIGHT_ENABLE
+    #undef RGBLED_NUM
+    #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/config.h:58]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/crkbd/keymaps/markstos/config.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#define TAPPING_TERM_THUMB 125
+
+// If you press a dual-role key, press another key, and then release the
+// dual-role key, all within the tapping term, by default the dual-role key
+// will perform its tap action. If the HOLD_ON_OTHER_KEY_PRESS option is
+// enabled, the dual-role key will perform its hold action instead.
+#define HOLD_ON_OTHER_KEY_PRESS
+
+// markstos: prevent keydown and keyup from firing on different layers
+#define PREVENT_STUCK_MODIFERS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/config.h:54]: This setting was deprecating a long time ago, and is the default behavior now.
```suggestion
```

File: keyboards/crkbd/keymaps/markstos/config.h [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+/* Select hand configuration */
+
+// #define MASTER_LEFT
+#define MASTER_RIGHT
+// #define EE_HANDS
+
+//#define SSD1306OLED
+
+#define USE_SERIAL_PD2
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/config.h:33]:
```suggestion
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 4]
```diff
@@ -0,0 +1,111 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+extern uint8_t is_master;
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/keymap.c:4]:
```suggestion
```

File: keyboards/crkbd/keymaps/markstos/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# markstos: Attempt to enable Bootmagic
+#BOOTMAGIC_ENABLE = lite
+
+# markstos: enable media keys
+EXTRAKEY_ENABLE = yes
+
+# markstos: smaller file size, little down-side
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/rules.mk:8]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/crkbd/keymaps/markstos/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+# markstos: for combos!
+COMBO_ENABLE = yes
+
+# markstos
+# Backlighting is not RGB underglow or RGB matrix
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/rules.mk:15]: As far as I'm aware, no version of the corne supports the backlight feature. 
```suggestion
```

File: keyboards/crkbd/keymaps/markstos/rules.mk [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+# This is for RGB *underglow*
+# https://github.com/qmk/qmk_firmware/blob/master/docs/feature_rgblight.md
+RGBLIGHT_ENABLE = no
+
+TAP_DANCE_ENABLE = no
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/rules.mk:21]: Nor tapdance.

File: keyboards/crkbd/keymaps/markstos/README.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Markstos Corne keyboard layout
```
Comment (drashna) [inline keyboards/crkbd/keymaps/markstos/README.md:1]: Please rename the file to be all lower case.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by markstos** [comment]
@drashna, thank you for the peer-review.

I have pushed fixup commits with the requested refinements and believe this branch is ready for a squash-and-merge.

Thanks.

**Review by drashna — APPROVED** [review]

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_HOME, KC_END , KC_MINS, KC_EQL , KC_PGDN,                      KC_LEFT, KC_DOWN, KC_UP  , KC_RGHT, KC_APP ,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_LT  , KC_GT  , KC_COPY, KC_PSTE, KC_SCLN,                      KC_MPLY, KC_MPRV, KC_MNXT, KC_VOLD, KC_VOLU,_______ ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          CTL_ESC, KC_TRNS, XXXXXXX,    RAISE  , KC_TRNS, KC_TRNS\
+                                      //`--------------------------'  `--------------------------'
+  ),
+
+  [_FUNC] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:87]:
```suggestion
  [_FUNC] = LAYOUT(
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+      _______, KC_1,    KC_2,   KC_3,    KC_4,    KC_5,                         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, XXXXXXX , KC_TILD,KC_GRV, KC_LBRC, KC_LCBR,                       KC_RCBR, KC_RBRC, KC_COMM,KC_DOT,  KC_SLSH, _______ ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          KC_TRNS,  KC_TRNS, LOWER,    KC_TRNS, KC_TRNS, KC_COLON \
+                                      //`--------------------------'  `--------------------------'
+    ),
+
+
+  [_RAISE] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:75]:
```suggestion
  [_RAISE] = LAYOUT(
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+ OSM(MOD_LALT),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                     KC_H    ,KC_J    ,KC_K    ,KC_L    ,KC_QUOT ,OSM_AGR ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+ OSM(MOD_LSFT),   KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                     KC_N    ,KC_M    ,KC_COMM ,KC_DOT  ,KC_SLSH ,OSL_FUN ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                         OSM_LCTL, GUI_ENT, LOW_TAB,   RSE_BSP ,KC_SPC  ,OSM_SFT \
+                                      //`--------------------------'  `--------------------------'
+  ),
+
+  [_LOWER] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:62]:
```suggestion
  [_LOWER] = LAYOUT(
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+#define LOW_TAB  LT(_LOWER, KC_TAB)
+#define RSE_BSP  LT(_RAISE, KC_BSPC)
+#define OSM_SFT  OSM(MOD_LSFT) 
+
+
+// For _RAISE layer
+#define CTL_ESC  LCTL_T(KC_ESC)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:50]:
```suggestion
  [_QWERTY] = LAYOUT(
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+                                      //`--------------------------'  `--------------------------'
+  ),
+
+  [_FUNC] = LAYOUT( \
+  //,-----------------------------------------------------.                    ,-----------------------------------------------------.
+      _______, KC_F1  , KC_F2  , KC_F3   , KC_F4 ,  KC_F5 ,                     KC_F6   , KC_F7  , KC_F8  , KC_F9  , KC_F10 ,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_F11 , KC_F12 , XXXXXXX, XXXXXXX, XXXXXXX,                     XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_CAPS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                     XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, RESET  ,XXXXXXX ,
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:93]:
```suggestion
      _______, KC_CAPS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                     XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, QK_BOOT,XXXXXXX ,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by markstos** [comment]
All suggested refinements have been accepted.

File: keyboards/crkbd/keymaps/markstos/rules.mk [hunk 15]
```diff
@@ -0,0 +1,14 @@
+# markstos: Attempt to enable Bootmagic
+#BOOTMAGIC_ENABLE = lite
+
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/rules.mk:3]:
```suggestion
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+  [_RAISE] = LAYOUT(
+  //,-----------------------------------------------------.                    ,-----------------------------------------------------.
+      _______, KC_DEL , XXXXXXX, KC_UNDS, KC_PLUS, KC_PGUP,                      XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS, KC_PIPE,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_HOME, KC_END , KC_MINS, KC_EQL , KC_PGDN,                      KC_LEFT, KC_DOWN, KC_UP  , KC_RGHT, KC_APP ,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_LT  , KC_GT  , KC_COPY, KC_PSTE, KC_SCLN,                      KC_MPLY, KC_MPRV, KC_MNXT, KC_VOLD, KC_VOLU,_______ ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          CTL_ESC, KC_TRNS, XXXXXXX,    RAISE  , KC_TRNS, KC_TRNS\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:83]:
```suggestion
                                          CTL_ESC, KC_TRNS, XXXXXXX,    RAISE  , KC_TRNS, KC_TRNS
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+  [_FUNC] = LAYOUT(
+  //,-----------------------------------------------------.                    ,-----------------------------------------------------.
+      _______, KC_F1  , KC_F2  , KC_F3   , KC_F4 ,  KC_F5 ,                     KC_F6   , KC_F7  , KC_F8  , KC_F9  , KC_F10 ,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_F11 , KC_F12 , XXXXXXX, XXXXXXX, XXXXXXX,                     XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,_______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_CAPS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                     XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, QK_BOOT,XXXXXXX ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          XXXXXXX, XXXXXXX, XXXXXXX,    XXXXXXX, FUNC   , XXXXXXX\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:95]:
```suggestion
                                          XXXXXXX, XXXXXXX, XXXXXXX,    XXXXXXX, FUNC   , XXXXXXX
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+  [_LOWER] = LAYOUT(
+  //,-----------------------------------------------------.                    ,-----------------------------------------------------.
+      _______, KC_EXLM, KC_AT,  KC_HASH, KC_DLR,  KC_PERC,                      KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, _______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, KC_1,    KC_2,   KC_3,    KC_4,    KC_5,                         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______ ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      _______, XXXXXXX , KC_TILD,KC_GRV, KC_LBRC, KC_LCBR,                       KC_RCBR, KC_RBRC, KC_COMM,KC_DOT,  KC_SLSH, _______ ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          KC_TRNS,  KC_TRNS, LOWER,    KC_TRNS, KC_TRNS, KC_COLON \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:70]:
```suggestion
                                          KC_TRNS,  KC_TRNS, LOWER,    KC_TRNS, KC_TRNS, KC_COLON
```

File: keyboards/crkbd/keymaps/markstos/keymap.c [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT(
+  //,-----------------------------------------------------.                    ,-----------------------------------------------------.
+       KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y    ,KC_U    ,KC_I    ,KC_O    ,KC_P    ,KC_DEL  ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+ OSM(MOD_LALT),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                     KC_H    ,KC_J    ,KC_K    ,KC_L    ,KC_QUOT ,OSM_AGR ,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+ OSM(MOD_LSFT),   KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                     KC_N    ,KC_M    ,KC_COMM ,KC_DOT  ,KC_SLSH ,OSL_FUN ,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                         OSM_LCTL, GUI_ENT, LOW_TAB,   RSE_BSP ,KC_SPC  ,OSM_SFT \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/markstos/keymap.c:58]:
```suggestion
                                         OSM_LCTL, GUI_ENT, LOW_TAB,   RSE_BSP ,KC_SPC  ,OSM_SFT
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by markstos** [comment]
@fauxpark , thank you for the peer-review.

I have pushed fixup commits with the requested refinements and believe this branch is ready for a squash-and-merge.

Thanks.

**Comment by markstos** [comment]
ping for review

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**